packages/fxa-settings/src/pages/ResetPassword/ResetPasswordWithRecoveryKeyVerified/container.tsx

```diff
@@ -55,11 +57,17 @@ const ResetPasswordWithRecoveryKeyVerifiedContainer = ({
   const navigateToHint = () => setShowHint(true);
   const navigateNext = async (continueToAccountEvent: () => void) => {
     if (isOAuthIntegration(integration)) {
+      // If we have lost the required bits for OAuth handling, we have to start again.
+      if (!sessionToken || !uid) {
+        navigate('/signin', { replace: true });
+        return;
+      }
+
       const { redirect, code, state, error } = await finishOAuthFlowHandler(
-        accountResetData.uid,
-        accountResetData.sessionToken,
-        accountResetData.keyFetchToken,
-        accountResetData.unwrapBKey
+        uid,
```

Comment:
This is the place where the compiler/IDE hints that this `uid` needs to be non-null so we have to figure out how to handle it before we get to this line. The fix above is a naive solution to the specific problem, happy to change it to something else.

Reply:
This is what the solution looks like today with the original STR.

https://github.com/user-attachments/assets/2ad9376b-324b-4e72-bd9e-4e9f96a0fb86